### PR TITLE
boot: zephyr: always clear SPLIM registers when present

### PR DESCRIPTION
**Pull request by hakonfam**
This fixes a bug where zephyr images would not boot correctly
on M33 cores.

PR-847 (commit 70af708b85b3) made clearing SPLIM registers
configurable. The registers would not be cleared if the architecture
had support for cleaning the state of the core before booting.

However, if the SPLIM register is not cleared before booting the
next image that image will get a fault when jumping to the routine
which cleans the core.

In other words we must clear the SPLIM registers when it is present.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

### CODE REVIEW

**Review by nvlsianpu — APPROVED** [review]

**Comment by hakonfam** [comment]
This will be resolved in zephyr instead.

https://github.com/zephyrproject-rtos/zephyr/pull/30190
